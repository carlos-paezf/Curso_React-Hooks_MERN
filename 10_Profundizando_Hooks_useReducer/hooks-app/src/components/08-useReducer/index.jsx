export { TodoApp } from "./TodoApp";
export { TodoList } from "./TodoList";
export { TodoListItem } from "./TodoListItem";
export { TodoAdd } from "./TodoAdd";