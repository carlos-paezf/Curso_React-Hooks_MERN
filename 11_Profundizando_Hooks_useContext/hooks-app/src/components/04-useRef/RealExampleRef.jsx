import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <>
            <h2>Real Example Ref</h2>
            <hr />
            {show && <MultipleCustomHooks />}

            <button className="btn btn-primary" onClick={() => setShow(!show)}>
                Show/Hide
            </button>
        </>
    );
};
