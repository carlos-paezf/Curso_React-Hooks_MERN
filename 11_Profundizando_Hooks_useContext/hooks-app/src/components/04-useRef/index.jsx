export { FocusScreen } from './FocusScreen'
export { RealExampleRef } from './RealExampleRef'