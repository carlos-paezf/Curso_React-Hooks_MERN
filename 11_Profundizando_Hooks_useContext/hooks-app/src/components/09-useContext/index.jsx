export { AppRouter } from "./AppRouter";

export { AboutScreen } from "./AboutScreen";
export { HomeScreen } from "./HomeScreen";
export { LoginScreen } from "./LoginScreen";
export { MainApp } from "./MainApp";

export { NavBar } from "./NavBar";