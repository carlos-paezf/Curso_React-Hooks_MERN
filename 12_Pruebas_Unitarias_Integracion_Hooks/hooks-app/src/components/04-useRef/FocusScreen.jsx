import React, { useRef } from 'react';

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        inputRef.current.focus()
        // document.querySelector('#input-focus').select()
    };
    

    return (
        <>
            <h2>Focus Screen</h2>
            <hr />
            <input ref={inputRef} id="input-focus" type="text" className="form-control" placeholder="Su nombre" />
            <button onClick={handleClick} className="btn btn-outline-primary mt-3">Focus</button>
        </>
    );
};
