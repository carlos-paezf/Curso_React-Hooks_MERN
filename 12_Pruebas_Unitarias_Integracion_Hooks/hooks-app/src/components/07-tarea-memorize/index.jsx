export { Hijo } from "./Hijo";
export { Padre } from "./Padre";