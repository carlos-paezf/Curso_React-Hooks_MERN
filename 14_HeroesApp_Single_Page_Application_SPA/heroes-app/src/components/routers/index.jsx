export { AppRouter } from './AppRouter'
export { DashboardRoutes } from './DashboardRoutes'