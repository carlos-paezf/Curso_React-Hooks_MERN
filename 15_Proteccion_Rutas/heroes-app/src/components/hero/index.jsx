export { HeroScreen } from './HeroScreen'
export { HeroList } from './HeroList'
export { HeroCard } from './HeroCard'