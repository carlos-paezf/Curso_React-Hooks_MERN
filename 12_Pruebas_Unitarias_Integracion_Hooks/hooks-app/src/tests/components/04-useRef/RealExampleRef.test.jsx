import React from 'react';
import { shallow } from 'enzyme'
import { RealExampleRef } from '../../../components/04-useRef';


describe('Pruebas al componente <RealExampleRef />', () => {

    let wrapper

    beforeEach(() => {
        wrapper = shallow(<RealExampleRef />)
    })

    /* The first test is a snapshot test. It checks that the component renders correctly. 
    The second test is a component test. It checks that the component renders a <MultipleCustomHooks /> component. */
    test('Debe mostrarse correctamenre', () => {
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false)
    })

    test('Debe de mostrar el componente <MultipleCustomHooks />', () => {
        wrapper.find('button').simulate('click')

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
    })
})
