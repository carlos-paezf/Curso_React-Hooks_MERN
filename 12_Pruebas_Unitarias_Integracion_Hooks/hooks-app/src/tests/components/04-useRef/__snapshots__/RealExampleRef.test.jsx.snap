// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <RealExampleRef /> Debe de mostrar el componente <MultipleCustomHooks /> 1`] = `
<Fragment>
  <h2>
    Real Example Ref
  </h2>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;

exports[`Pruebas al componente <RealExampleRef /> Debe mostrarse correctamenre 1`] = `
<Fragment>
  <h2>
    Real Example Ref
  </h2>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
